node/core/pvf/src/prepare/worker.rs

```diff
@@ -294,5 +374,50 @@ fn prepare_artifact(code: &[u8]) -> Result<CompiledArtifact, PrepareError> {
 	.map_err(|panic_payload| {
 		PrepareError::Panic(crate::error::stringify_panic_payload(panic_payload))
 	})
-	.and_then(|inner_result| inner_result)
+	.and_then(|inner_result| inner_result)?;
+
+	let cpu_time_elapsed = cpu_time_start.elapsed();
+
+	let lock_result = lock.compare_exchange(false, true, Ordering::Relaxed, Ordering::Relaxed);
+	if lock_result.is_err() {
+		// The other thread is still sending an error response over the socket. Wait on it and
+		// return.
+		match handle.join() {
+			Ok(()) => {
+				unreachable!(
```

Comment:
This unreachable looks really scary and unnecessary, I would've went with a simple
```rust
// Monitor thread detected timeout and likely already terminated a process, nothing to do.
return
```

Reply:
We can also have a log or even a debug_assert.

Reply:
Hmm, I don't see the point of logging a case that's unreachable, which we have a proof for. I would personally keep the unreachable and maybe just make it less verbose. We also can't just do an empty return here since `worker_event_loop` expects an `io::Result<Never>>`.

Reply:
Yes, `continue` instead. Or
> make it less verbose

would also work

Reply:
Makes sense, removed the `unreachable!`.